### DISCUSSION TITLE
请问下如何捕获robot重连异常

### DISCUSSION THREAD

**Discussion by mahaonanStart** [mamoe/mirai#909] [Q&A] [answered]
java如何捕获重连异常，因为我看重连次数默认设置的Intege.MAX，但是我机器上只要掉线，重连会一直失败，只能重启。因此我想捕获重连异常，然后调用脚本重启程序。但是不知道怎么捕捉这个异常？求助各位

版本是最新版本2.1.0

**Comment by Him188** [comment, marked as answer]
可以尝试 2.0.0, 这个异常不能捕捉

**Reply by mahaonanStart** [reply]
之前的版本也是，只要重连就是一直失败，超时异常，可不可以开放这个异常的捕捉呢，这样有啥问题也可以第一时间自动重启

**Reply by Him188** [reply]
可以监听BotOfflineEvent, 如果长时间没有 BotReloginEvent, 就说明重连失败了
